pkg/controllers/tikv/tasks/offline.go

```diff
@@ -23,23 +23,34 @@ import (
 )
 
 const (
-	// removingWaitInterval is the interval to wait for store removal.
-	removingWaitInterval = 10 * time.Second
+	defaultLeaderEvictTimeout = 5 * time.Minute
+	jitter                    = 10 * time.Second
 )
 
 func TaskOfflineStore(state *ReconcileContext) task.Task {
 	return task.NameTaskFunc("OfflineStore", func(ctx context.Context) task.Result {
-		if state.Store == nil {
-			return task.Complete().With("store is not exists, no need to offline")
+		if state.GetStoreState() == v1alpha1.StoreStateRemoving {
+			return task.Wait().With("the store is removing")
 		}
 
-		if state.PDClient == nil {
-			return task.Fail().With("pd client is not registered")
+		var reason string
+		delTime := state.Object().GetDeletionTimestamp()
+		switch {
+		// leaders evicted
+		case state.LeaderEvicting && state.GetLeaderCount() == 0:
+			reason = "leaders have been all evicted"
+		case !delTime.IsZero() && delTime.Add(defaultLeaderEvictTimeout).Before(time.Now()):
+			reason = "leader eviction timeout"
```

Comment:
Use the constant defined in `api`? It seems it's duplicated with `syncLeadersEvictedCond`?